topi/python/topi/x86/nn.py

```diff
@@ -54,3 +56,61 @@ def schedule_pool(outs):
         The computation schedule for the op.
     """
     return _default_schedule(outs, False)
+
+
+@generic.schedule_dense.register(["cpu"])
+def schedule_dense(outs):
+    """Schedule for dense
+
+    Parameters
+    ----------
+    outs: Array of Tensor
+          The computation graph description of pool
+          in the format of an array of tensors.
+
+    Returns
+    -------
+    sch: Schedule
+        The computation schedule for the op.
+    """
+
+    outs = [outs] if isinstance(outs, tvm.tensor.Tensor) else outs
+    s = tvm.create_schedule([x.op for x in outs])
+
+    def traverse(op):
+        """Traverse operators from computation graph"""
+        # inline all one-to-one-mapping operators except the last stage (output)
+        if tag.is_broadcast(op.tag):
+            if op not in s.outputs:
+                s[op].compute_inline()
+            for tensor in op.input_tensors:
+                if tensor.op.input_tensors:
+                    traverse(tensor.op)
+
+        if 'dense' in op.tag:
+            C = op.output(0)
+            x, y = C.op.axis
+
+            # Write cache for blocks
+            CC = s.cache_write(C, 'global')
+
+            # Tile
+            bnx = 1
+            bny = 4
+            _, yo, _, yi = s[C].tile(x, y, bnx, bny)
+            s[CC].compute_at(s[C], yo)
+            xc, yc = s[CC].op.axis
+            k, = s[CC].op.reduce_axis
+            ko, ki = s[CC].split(k, factor=4)
+            s[CC].reorder(ko, xc, ki, yc)
+            s[CC].unroll(ki)
+            s[CC].vectorize(yc)
```

Comment:
will it be better if we have vec_len > 4, say, 8 or 16 for avx2/avx512?

Reply:
In this schedule, increasing bny will result in worse performance for avx2/avx512.

Reply:
ki is not the last index and is going to be unrolled, plus the vectorization happens at yc, so I guess it should be fine.

Reply:
oh, just realized that you are talking about bny. If the GEMM is [1, 25088] * [25088, 4096] in row major, tiling it using bny makes sense because the reduce axis is already too large. Perhaps we should pack B in a better way.

Reply:
Also, if A only has only one row, tiling C doesn't make the procedure any different.

Reply:
Yes. I tried the packing method shown in tvm gemm tutorial, but it gave much worse performance. Next step I'll investigate to way mkldnn doing packing. 
 The reason for tiling C is to provide a generic(potentially) interface to cover the case in which  batch size > 1. For now we can also remove that and just split y axis.